### ISSUE TITLE
[RayService][Bug] when doing multi-node serving with vLLM, non-primary worker pods reports unready. 

### ISSUE THREAD

**Issue by kanwang** [ray-project/kuberay#2552]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

I followed https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/vllm/ray-service.vllm.yaml to setup vllm serving with RayService. While it works out, I see an issue when enabling multi-node inference.  

I configured `PIPELINE_PARALLELISM = 2` and the service started up correctly. But the second pod always report `Ready = False`. Looks like with this example, the Deployment is only deployed to one pod, and under the hood vLLM leverages the second pod for distributed inference. Thus, the Proxy isn't deployed to second pod and readiness check isn't passing. 

Looks like the behavior was introduced by https://github.com/ray-project/kuberay/pull/1808

While this isn't an immediate problem (serving is working as expected), it doesn't seem right to report unready pods while it actually is ready. In addition, unready pods has other infrastructure impact on our k8s cluster (e.g. impacting node lifecycle management). 

### Reproduction script

Following https://docs.ray.io/en/latest/cluster/kubernetes/examples/vllm-rayservice.html or https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/vllm/ray-service.vllm.yaml and set `PIPELINE_PARALLELISM=2` will reproduce this issue. 

### Anything else

* It is always re-producible. 

* I don't have any solution in mind. but I am happy to brainstorm together and implement/tryout any solution if there is a suggestion! 

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by andrewsykim** [comment]
On Slack you mentioned overriding the readiness probe and then using proxy like Envoy. This could work but I think the implementation will be tricky to get right because it's not trivial to know which Pods are running the proxy actors without connecting to the Ray Cluster. Another workaround is potentially just running a dummy proxy actor on every node?

**Comment by kanwang** [comment]
> potentially just running a dummy proxy actor on every node

yes that's another idea I thought about. basically a dummy proxy that just route traffic? I was actually hoping this config: https://docs.ray.io/en/latest/serve/api/doc/ray.serve.config.ProxyLocation.html to do that. but when I was testing it, it seems to be ignoring the second pod vLLM was using - it didn't deploy proxy on that pod. 
Is there a different way to start proxy actor everywhere?

**Comment by andrewsykim** [comment]
Can you just deploy another Ray Serve Deployment that is doing nothing? You may need to add enough replicas / resources to ensure it runs on every node

**Comment by kevin85421** [comment]
Ray Serve seems not providing to deploy proxy actor on every node although there are no replicas on it.

![image](https://github.com/user-attachments/assets/e397ffd3-afe6-4373-80c1-377528dd932f)

**Comment by kanwang** [comment]
Yes, Ray Serve doesn't support this natively. so what I ended up doing is deployment a `VLLMDeployment` and a `VLLMProxy` as 2 deployments in one app, deploy the `VLLMProxy` to enough replicas so it's deployed on all nodes, and then use [model composition](https://docs.ray.io/en/latest/serve/model_composition.html) to route request to the VLLM deployment. 

Something like this:
```
@serve.deployment()
@serve.ingress(app)
class VLLMProxy:
    def __init__(self, vllm_processor: DeploymentHandle):
        self.vllm_processor = vllm_processor
        self.stream_vllm_processor = vllm_processor.options(stream=True)

    @app.get("/v1/models")
    async def list_models(self):
        content =  await self.vllm_processor.list_models.remote()
        return JSONResponse(content=content)

    @app.post("/v1/chat/completions")
    async def create_chat_completion(self, request: ChatCompletionRequest, raw_request: Request):
        generator = self.stream_vllm_processor.create_chat_completion.remote(request)
        if isinstance(generator, ErrorResponse):
            return JSONResponse(
                content=generator.model_dump(), status_code=generator.code
            )
        if request.stream:
            return StreamingResponse(content=generator, media_type="text/event-stream")

        assert isinstance(generator, ChatCompletionResponse)
        return JSONResponse(content=generator.model_dump())

```


So maybe this isn't a bug for rayservice, but it definitely is a broken experience when Ray pods report unready.
